plugins/BEdita/Core/tests/TestCase/ORM/Inheritance/InheritanceEventHandlerTest.php

```diff
@@ -309,6 +311,53 @@ public function testBeforeSaveFailure()
     }
 
     /**
+     * Test options passed in che chain
+     *
+     * @return void
+     *
+     * @covers ::beforeSave()
+     */
+    public function testBeforeSaveOptions()
+    {
+        // Main table
+        $this->fakeFelines->eventManager()->on(
+            'Model.beforeSave',
+            function (Event $event, EntityInterface $entity, \ArrayObject $options) {
+                static::assertArrayNotHasKey('_inherited', $options);
+                static::assertTrue($options['atomic']);
+            }
+        );
+
+        // Inherited table
+        $this->fakeMammals->eventManager()->on(
+            'Model.beforeSave',
+            function (Event $event, EntityInterface $entity, \ArrayObject $options) {
+                static::assertArrayHasKey('_inherited', $options);
+                static::assertTrue($options['_inherited']);
+                static::assertFalse($options['atomic']);
+            }
+        );
+
+        // Inherited table
+        $this->fakeAnimals->eventManager()->on(
+            'Model.beforeSave',
+            function (Event $event, EntityInterface $entity, \ArrayObject $options) {
+                static::assertArrayHasKey('_inherited', $options);
+                static::assertTrue($options['_inherited']);
+                static::assertFalse($options['atomic']);
+            }
+        );
+
+        $feline = $this->fakeFelines->newEntity([
+            'name' => 'Gastone',
+            'legs' => 4,
+            'subclass' => 'Lucky pets',
+            'family' => 'Cats',
+        ]);
+        $result = $this->fakeFelines->save($feline);
+    }
```

Comment:
If for some reason some of the listened events were not dispatched, this test would not fail. Adding something like this would be enough:

```php
$eventDispatchedFelines = $eventDispatchedMammals = /* ... */ = 0;

$this->fakeFelines->eventManager()->on(
    'Model.beforeSave',
    function (Event $event, EntityInterface $entity, \ArrayObject $options) use (&$eventDispatchedFelines) {
        $eventDispatchedFelines++;
        // ...
    }
);

// ...

$result = $this->fakeFelines->save($feline);

static::assertSame(1, $eventDispatchedFelines);
static::assertSame(1, $eventDispatchedMammals);
// ...
```

Reply:
You are right.

Reply:
done